# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/dns/DNSPacket.h
# Commit: 6f4ca967aa44
# Full Hash: 6f4ca967aa446cf24f4224576f51803c982f3d7b
# Author: Kershaw Chang <kershaw@mozilla.com>
# Date: 2024-06-14 20:49:45
# Regressor Bug: 1901990
# File Overlap Count: 2
# Description:
#   Bug 1901990 - Introduce a pref for setting a mock HTTPS RR, r=necko-reviewers,valentin
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D213559
# ==============================================================================

diff -r 8a2bdb44382a -r 6f4ca967aa44 netwerk/dns/DNSPacket.h
--- a/netwerk/dns/DNSPacket.h	Fri Jun 14 13:27:36 2024 +0000
+++ b/netwerk/dns/DNSPacket.h	Fri Jun 14 13:42:37 2024 +0000
@@ -52,6 +52,8 @@
   nsresult OnDataAvailable(nsIRequest* aRequest, nsIInputStream* aInputStream,
                            uint64_t aOffset, const uint32_t aCount);
 
+  // Encode the input host name into a sequence of labels.
+  static nsresult EncodeHost(nsCString& aBody, const nsACString& aHost);
   // Encodes the name request into a buffer that represents a DNS packet
   virtual nsresult EncodeRequest(nsCString& aBody, const nsACString& aHost,
                                  uint16_t aType, bool aDisableECS);